MapboxDirections/Match/MBTracepoint.swift

```diff
@@ -11,16 +11,16 @@ public class Tracepoint: Waypoint {
     /**
      Index of the waypoint inside the matched route.
      */
-    open var waypointIndex: Int?
+    @objc open var waypointIndex: Int = NSNotFound
     
     /**
-      Index to the match object in matchings the sub-trace was matched to.
+      If more than one `Match` are returned, this index represents which `Match` this tracepoint was used on.
```

Comment:
This is a very poor API. The tracepoint object has an index into an arbitrary array of match objects. What if client code rearranges the array or creates a new array? Then this index is meaningless but is still set.

This suggests very strongly that `MatchCompletionHandler`’s first argument (`tracepoints`) should be removed, and instead Match should have a property containing an array of tracepoints.